sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala

```diff
@@ -115,12 +115,12 @@ class IntervalUtilsSuite extends SparkFunSuite {
     checkFromInvalidString("2234567890 days", "integer overflow")
     checkFromInvalidString("\n", "Error parsing '\n' to interval")
     checkFromInvalidString("\t", "Error parsing '\t' to interval")
-
+    checkFromInvalidString("1. seconds", "invalid value '1.'")
+    checkFromInvalidString(". seconds", "invalid value '.'")
   }
 
   test("string to interval: seconds with fractional part") {
     checkFromString("0.1 seconds", new CalendarInterval(0, 0, 100000))
-    checkFromString("1. seconds", new CalendarInterval(0, 0, 1000000))
```

Comment:
Will we explicitly disable this case? 

```scala
scala> sql("select 0.")
res0: org.apache.spark.sql.DataFrame = [0: decimal(1,0)]
```

At least I know Python, Java, Scala and R support this way as well.

Reply:
```sql
presto> select interval '1.' second;
     _col0
----------------
 0 00:00:01.000
(1 row)

Query 20191119_051438_00001_f5kcs, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
0:03 [0 rows, 0B] [0 rows/s, 0B/s]

presto> select interval '.' second;
Query 20191119_051452_00002_f5kcs failed: Invalid INTERVAL SECOND value: .
```
Also check with presto, '1.' seconds is valid

Reply:
@HyukjinKwon good point. To be consistent with Spark's own parser, maybe we should allow `interval '1.' second`. But `interval '.' second` should be invalid.

Reply:
^ +1!

Reply:
updated, please recheck, thanks.